File: src/main/java/com/mbayou/kick4k/events/type/ChannelSubscriptionRenewalEvent.java
package com.mbayou.kick4k.events.type;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;

public class ChannelSubscriptionRenewalEvent extends KickEvent {
    private final EventUser broadcaster;
    private final EventUser subscriber;
    private final Integer duration;
    private final Instant createdAt;
    private final Instant expiresAt;

    @JsonCreator
    public ChannelSubscriptionRenewalEvent(@JsonProperty("broadcaster") EventUser broadcaster,
                                           @JsonProperty("subscriber") EventUser subscriber,
                                           @JsonProperty("duration") Integer duration,
                                           @JsonProperty("created_at") Instant createdAt,
                                           @JsonProperty("expires_at") Instant expiresAt) {
        this.broadcaster = broadcaster;
        this.subscriber = subscriber;
        this.duration = duration;
        this.createdAt = createdAt;
        this.expiresAt = expiresAt;
    }

    public EventUser getBroadcaster() {
        return broadcaster;
    }

    public EventUser getSubscriber() {
        return subscriber;
    }

    public Integer getDuration() {
        return duration;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getExpiresAt() {
        return expiresAt;
    }

    public static String getEventType() {
        return "channel.subscription.renewal";
    }

    public static String getEventVersion() {
        return "1";
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/type/ChannelSubscriptionRenewalEvent.kt
package com.mbayou.kick4k.events.type

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant

data class ChannelSubscriptionRenewalEvent @JsonCreator constructor(
    @JsonProperty("broadcaster") val broadcaster: EventUser?,
    @JsonProperty("subscriber") val subscriber: EventUser?,
    @JsonProperty("duration") val duration: Int?,
    @JsonProperty("created_at") val createdAt: Instant?,
    @JsonProperty("expires_at") val expiresAt: Instant?,
) : KickEvent() {
    companion object {
        @JvmStatic
        fun getEventType(): String = "channel.subscription.renewal"

        @JvmStatic
        fun getEventVersion(): String = "1"
    }
}
